graphqlapi/src/main/java/io/stargate/graphql/web/resources/GraphqlCache.java

```diff
@@ -141,6 +142,23 @@ public GraphQL getDml(
     return currentHolder == null ? null : currentHolder.getGraphql();
   }
 
+  public void putDml(
+      String keyspaceName, SchemaSource newSource, GraphQL graphql, AuthenticationSubject subject) {
+    Map<String, String> headers = subject.customProperties();
+    DataStore dataStore = buildUserDatastore(subject, headers);
+    String decoratedKeyspaceName = persistence.decorateKeyspaceName(keyspaceName, headers);
+
+    Keyspace keyspace = dataStore.schema().keyspace(keyspaceName);
+    if (keyspace == null) {
+      LOG.trace("Keyspace {} does not exist", decoratedKeyspaceName);
+      return;
+    }
```

Comment:
I think we can skip that check. We're coming from the deploy schema fetcher that already retrieved the keyspace in order to build the `GraphQL`.
That way you don't need to instantiate a `DataStore` either.

Granted, the keyspace could have been dropped in the meantime, but this can already happen anyway. The schema-first API is not completely immune to external schema changes (that would be nearly impossible to do since there is no way to lock the schema).

Reply:
Yes, from the current context it should not happen. However, the `putDml` is public on the `GraphqlCache`, therefore it can be used by other places in code in the future. In that case, I think that guarding against this situation may be beneficial, wdyt?

Reply:
Even if calling from another place, you'll need to already have the `GraphQL` instance, so that implies you've built it.